tarofreighter/chain_porter.go

```diff
@@ -515,42 +515,33 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 func (p *ChainPorter) advanceState(pkg *sendPackage,
 	txBroadcastRespChan chan *PendingParcel) error {
 
-	// Some states require a goroutine because tx confirmation and proof
-	// transfer can take a while.
-	p.Wg.Add(1)
-	go func(pkg *sendPackage) {
-		defer p.Wg.Done()
-
-		// Continue state transitions whilst state complete has not yet
-		// been reached.
-		for pkg.SendState < SendStateComplete {
-			log.Infof("ChainPorter executing state: %v",
-				pkg.SendState)
-
-			// Before we attempt a state transition, make sure that
-			// we aren't trying to shut down.
-			select {
-			case <-p.Quit:
-				err := fmt.Errorf("porter shutting down")
-				p.cfg.ErrChan <- err
-				break
-
-			default:
-			}
+	// Continue state transitions whilst state complete has not yet
+	// been reached.
+	for pkg.SendState < SendStateComplete {
+		log.Infof("ChainPorter executing state: %v",
+			pkg.SendState)
+
+		// Before we attempt a state transition, make sure that
+		// we aren't trying to shut down.
+		select {
+		case <-p.Quit:
+			return nil
```

Comment:
Ah I see, it's changed here. This is probably an instance of one too many commits, would make it easier to understand if this was perhaps done earlier or squashed with the one that made the change with the error channel.